internal/controller/promotions/promotions.go

```diff
@@ -379,6 +379,7 @@ func (r *reconciler) promote(
 				err,
 			)
 		}
+		newStatus.PromotedFreight = &nextFreight
```

Comment:
I believe we could also benefit from having this reference _before_ the promotion in full is actually successful.

For example, to be able to facilitate #1613 in which the Git promotion may produce a "health check commit" which we later want to use to ensure an Argo CD Application has successfully been synced.

Reply:
Agreed. This should be captured at the earliest opportunity.